cpp/src/io/avro/avro_gpu.cu

```diff
@@ -190,7 +197,44 @@ avro_decode_row(schemadesc_s const* schema,
           skip += schema[i].count;  // Should always be 1
         }
       } break;
+
+      case type_duration: {
+        // A duration logical type annotates Avro fixed type of size 12, which
+        // stores three little-endian unsigned integers that represent durations
+        // at different granularities of time. The first stores a number in
+        // months, the second stores a number in days, and the third stores a
+        // number in milliseconds.
+        CUDF_UNREACHABLE("avro type 'duration' not yet implemented");
```

Comment:
`CUDF_UNREACHABLE` seems incorrect to me. If this is a feature we do not yet support then I think `CUDF_FAIL` would be more appropriate.
https://github.com/rapidsai/cudf/blob/branch-23.04/cpp/doxygen/developer_guide/DEVELOPER_GUIDE.md#runtime-conditions
https://github.com/rapidsai/cudf/blob/0ee82c9619ef2145ea270c6c97eb5c5060eb2884/cpp/include/cudf/detail/utilities/assert.cuh#L40-L43

Reply:
Ah, this is device code though, so:
```
/home/trent/src/cudf/cpp/src/io/avro/avro_gpu.cu(207): error: device code does not support exception handling
```
Originally these were just `// XXX TODO` comments, and @vuule suggested using `CUDF_UNREACHABLE`: https://github.com/rapidsai/cudf/pull/12788#discussion_r1122442063

Reply:
Ah very good. I got lost and did not realize this was device code.